src/stats.ml

```diff
@@ -67,3 +75,10 @@ let end_replace ~sampling_interval =
     stats.replace_times <- average :: stats.replace_times;
     replace_timer := Mtime_clock.counter ();
     nb_replace := 0 )
+
+let sync_with_timer f =
+  let timer = Mtime_clock.counter () in
+  f ();
+  let span = Mtime_clock.count timer in
+  let time = Mtime.Span.to_us span in
+  stats.sync_times <- time :: stats.sync_times
```

Comment:
This looks like an unbounded list and should probably be documented somewhere

Reply:
Replaced this with a timer for only the latest sync call (instead of storing all previous sync calls). Also, by default every call to sync is timed now, so I removed the `sync_with_timer`.